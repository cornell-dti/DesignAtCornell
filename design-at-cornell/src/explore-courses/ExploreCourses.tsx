import React, { useState } from 'react';
import Title from './title/Title';
import {
  PageContainer,
  Sort,
  CourseGrid,
  CourseBubble,
  TagsContainer,
  Tag
} from './ExploreCoursesStyles';

type Course = {
  courseId: string;
  courseTitle: string;
  tags: string[];
};

const ExploreCourses = () => {
  const [courses, setCourses] = useState<Course[]>([
    {courseId: "INFO 1300", courseTitle: "Introductory Design and Programming for the Web", tags: ["Information Science", "3 Credits"]},
    {courseId: "DEA 1500", courseTitle: "Introduction to Environmental Psychology", tags: ["Design and Enviro...", "3 Credits"]},
    {courseId: "FSAD 1120", courseTitle: "Fashion Design and Visual Thinking", tags: ["Fashion Design an...", "Fall"]},
    {courseId: "ART 2201", courseTitle: "Painting: Introduction to Painting", tags: ["Fine Arts", "3 Credits"]},
    {courseId: "ARCH 1101", courseTitle: "Design I", tags: ["Architecture", "3 Credits", "Fall"]},
    {courseId: "ART 2301", courseTitle: "Print Media: Introduction to Print Media", tags: ["Fine Arts", "3 Credits"]},
  ]);

  const courseBubbles = (
    <CourseGrid>
      {courses.map(course => (
        <CourseBubble>
          <h6>{course.courseId}</h6>
          <p>{course.courseTitle}</p>
          <TagsContainer>
            {course.tags.map(tag => (
              <Tag>
                <p>{tag}</p>
              </Tag>
            ))}
          </TagsContainer>
        </CourseBubble>
      ))}
    </CourseGrid>
  );

  return (
    <PageContainer>
    <Title />
    <Sort>
      <p>Sort By</p>
    </Sort>
    {courseBubbles}
  </PageContainer>
  );
};

export default ExploreCourses;