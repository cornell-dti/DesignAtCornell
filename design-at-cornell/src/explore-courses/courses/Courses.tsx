import React from 'react';
import { PageContainer, CourseGrid } from '../ExploreCoursesStyles';
import Sort from './Sort';
import CourseBubble from './CourseBubble';
import { CoursesProps } from '../types/PropertyTypes'
import Category, { getSelections } from '../types/Category';

const Courses = ({ courses, filterData }: CoursesProps) => {

  const selectedDesignAreas = getSelections(filterData, Category['Design Areas']);
  const selectedMajors = getSelections(filterData, Category['Majors/Minors']);
  const selectedSemesters = getSelections(filterData, Category['Semester']);
  const selectedCredits = getSelections(filterData, Category['Credits']);
  const selectedLevels = getSelections(filterData, Category['Level']);

  const courseBubbles = (
    <CourseGrid>
      {courses
        .filter(({ code, content }) => {
          const {
             designAreas, major, semester, credits
          } = content;
          if (selectedDesignAreas.size > 0 && !designAreas.some(area => selectedDesignAreas.has(area))) return false;
          if (selectedMajors.size > 0 && !selectedMajors.has(major)) return false;
          if (selectedSemesters.size > 0 && !semester.some(str => selectedSemesters.has(str))) return false;
          const levelString = (code - code % 1000).toString();
          if (selectedLevels.size > 0 && !selectedLevels.has(levelString)) return false;
          const creditsString = credits < 5 ? credits.toString() : '5+';
          return selectedCredits.size === 0 || selectedCredits.has(creditsString);
        })
        .map(course => (
          <CourseBubble
            key={course.id + ' ' + course.code}
            {...course}
          />
        ))
      }
    </CourseGrid>
  );

  return (
    <PageContainer>
      <Sort />
      {courseBubbles}
    </PageContainer>
  );
};

export default Courses;
