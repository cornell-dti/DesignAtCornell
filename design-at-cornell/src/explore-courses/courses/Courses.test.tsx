import React from 'react';
import renderer from 'react-test-renderer';
import Courses from './Courses';
import { Course } from '../ExploreCourses';
import 'jest-styled-components';

describe('<CourseBubble />', () => {
  it('renders correctly for no courses', () => {
    const courses: Course[] = []

    const tree = renderer
      .create(<Courses {...courses}/>)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('renders correctly for one course', () => {
    const courses: Course[] = [
      {courseCode: 'TEST1110', courseTitle: 'Testing Coruse 1110', tags: ['tag 1', 'tag 2']}
    ]

    const tree = renderer
      .create(<Courses {...courses}/>)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('renders correctly for 8 course', () => {
    const courses: Course[] = [
      {courseCode: 'TEST1110', courseTitle: 'Testing Coruse 1110', tags: ['tag 1', 'tag 2']},
      {courseCode: 'TEST1111', courseTitle: 'Testing Coruse 1111', tags: ['tag 1', 'tag 3']},
      {courseCode: 'TEST1112', courseTitle: 'Testing Coruse 1112', tags: ['tag 3', 'tag 4']},
      {courseCode: 'TEST1113', courseTitle: 'Testing Coruse 1113', tags: ['tag 5', 'tag 6']},
      {courseCode: 'TEST1114', courseTitle: 'Testing Coruse 1114', tags: ['tag 5', 'tag 7', 'tag 9']},
      {courseCode: 'TEST1115', courseTitle: 'Testing Coruse 1115', tags: ['tag 6']},
      {courseCode: 'TEST1116', courseTitle: 'Testing Coruse 1116', tags: ['tag 6', 'tag 8']},
      {courseCode: 'TEST1117', courseTitle: 'Testing Coruse 1117', tags: []}
    ]

    const tree = renderer
      .create(<Courses {...courses}/>)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
