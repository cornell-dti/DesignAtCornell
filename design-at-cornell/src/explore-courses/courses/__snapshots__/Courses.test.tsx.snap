// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Courses /> renders correctly for 3 course 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  width: 100%;
  height: 100%;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  width: 100%;
  height: 80px;
  padding-top: 20px;
  padding-right: 80px;
  -webkit-align-items: flex-end;
  -webkit-box-align: flex-end;
  -ms-flex-align: flex-end;
  align-items: flex-end;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  width: -webkit-fit-content;
  width: -moz-fit-content;
  width: fit-content;
  height: 100%;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c2 h6 {
  font-weight: bold;
  font-size: 20px;
  line-height: 23px;
  margin: 0;
  color: black;
}

.c2 p {
  font-weight: bold;
  font-size: 20px;
  line-height: 23px;
  margin: 0;
  color: #828282;
}

.c2 img {
  width: 12px;
  height: 12px;
  margin-left: 10px;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  width: 100%;
  height: -webkit-fit-content;
  height: -moz-fit-content;
  height: fit-content;
  padding: 0 80px;
}

.c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  width: 22%;
  min-width: 250px;
  height: 197px;
  cursor: pointer;
  background: white;
  border: 1px solid;
  border-radius: 8px;
  margin: 1.5%;
  padding: 25px;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.c6 {
  font-weight: bold;
  font-size: 20px;
  line-height: 23px;
  text-align: start;
  margin: 0;
  color: black;
}

.c5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  width: 100%;
  height: -webkit-fit-content;
  height: -moz-fit-content;
  height: fit-content;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.c5 p {
  font-weight: bold;
  font-size: 16px;
  line-height: 19px;
  text-align: start;
  margin: 0;
  color: #898989;
}

.c5 img {
  cursor: pointer;
}

.c7 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-flex-wrap: nowrap;
  -ms-flex-wrap: nowrap;
  flex-wrap: nowrap;
  overflow: hidden;
  width: 100%;
  height: -webkit-fit-content;
  height: -moz-fit-content;
  height: fit-content;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
}

.c8 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: -webkit-fit-content;
  width: -moz-fit-content;
  width: fit-content;
  height: 28px;
  border-radius: 43px;
  margin-right: 10px;
  padding-left: 10px;
  padding-right: 10px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: rgba(255,255,255,0.53);
  white-space: nowrap;
  font-weight: bold;
  font-size: 12px;
  color: black;
}

<div
  className="c0"
>
  <div
    className="c1"
  >
    <div
      className="c2"
    >
      <h6>
        Sort By  
      </h6>
      <p>
        recommended
      </p>
      <img
        alt="sort"
        src="down-arrow.png"
      />
    </div>
  </div>
  <div
    className="c3"
  >
    <div
      className="c4"
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      style={
        Object {
          "borderColor": "#C9E1AA",
        }
      }
    >
      <div
        className="c5"
      >
        <p>
          TEST 1101
        </p>
        <img
          alt="save course"
          src="bookmark.svg"
        />
      </div>
      <div
        className="c6"
      >
        Intro to Design
      </div>
      <div
        className="c7"
      >
        <div
          className="c8"
          style={
            Object {
              "background": "#BAE6FF",
            }
          }
        >
          4 Credits
        </div>
        <div
          className="c8"
          style={
            Object {
              "background": "#FFE5A1",
            }
          }
        >
          Fall, Spring
        </div>
      </div>
    </div>
    <div
      className="c4"
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      style={
        Object {
          "borderColor": "#FFCBBA",
        }
      }
    >
      <div
        className="c5"
      >
        <p>
          TEST 1102
        </p>
        <img
          alt="save course"
          src="bookmark.svg"
        />
      </div>
      <div
        className="c6"
      >
        Intro to Design
      </div>
      <div
        className="c7"
      >
        <div
          className="c8"
          style={
            Object {
              "background": "#BAE6FF",
            }
          }
        >
          4 Credits
        </div>
        <div
          className="c8"
          style={
            Object {
              "background": "#FFE5A1",
            }
          }
        >
          Fall, Spring
        </div>
      </div>
    </div>
    <div
      className="c4"
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      style={
        Object {
          "borderColor": "#D3D2F7",
        }
      }
    >
      <div
        className="c5"
      >
        <p>
          TEST 1103
        </p>
        <img
          alt="save course"
          src="bookmark.svg"
        />
      </div>
      <div
        className="c6"
      >
        Intro to Design
      </div>
      <div
        className="c7"
      >
        <div
          className="c8"
          style={
            Object {
              "background": "#BAE6FF",
            }
          }
        >
          4 Credits
        </div>
        <div
          className="c8"
          style={
            Object {
              "background": "#FFE5A1",
            }
          }
        >
          Fall, Spring
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`<Courses /> renders correctly for no courses 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  width: 100%;
  height: 100%;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  width: 100%;
  height: 80px;
  padding-top: 20px;
  padding-right: 80px;
  -webkit-align-items: flex-end;
  -webkit-box-align: flex-end;
  -ms-flex-align: flex-end;
  align-items: flex-end;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  width: -webkit-fit-content;
  width: -moz-fit-content;
  width: fit-content;
  height: 100%;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c2 h6 {
  font-weight: bold;
  font-size: 20px;
  line-height: 23px;
  margin: 0;
  color: black;
}

.c2 p {
  font-weight: bold;
  font-size: 20px;
  line-height: 23px;
  margin: 0;
  color: #828282;
}

.c2 img {
  width: 12px;
  height: 12px;
  margin-left: 10px;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  width: 100%;
  height: -webkit-fit-content;
  height: -moz-fit-content;
  height: fit-content;
  padding: 0 80px;
}

<div
  className="c0"
>
  <div
    className="c1"
  >
    <div
      className="c2"
    >
      <h6>
        Sort By  
      </h6>
      <p>
        recommended
      </p>
      <img
        alt="sort"
        src="down-arrow.png"
      />
    </div>
  </div>
  <div
    className="c3"
  />
</div>
`;

exports[`<Courses /> renders correctly for one course 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  width: 100%;
  height: 100%;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  width: 100%;
  height: 80px;
  padding-top: 20px;
  padding-right: 80px;
  -webkit-align-items: flex-end;
  -webkit-box-align: flex-end;
  -ms-flex-align: flex-end;
  align-items: flex-end;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  width: -webkit-fit-content;
  width: -moz-fit-content;
  width: fit-content;
  height: 100%;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c2 h6 {
  font-weight: bold;
  font-size: 20px;
  line-height: 23px;
  margin: 0;
  color: black;
}

.c2 p {
  font-weight: bold;
  font-size: 20px;
  line-height: 23px;
  margin: 0;
  color: #828282;
}

.c2 img {
  width: 12px;
  height: 12px;
  margin-left: 10px;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  width: 100%;
  height: -webkit-fit-content;
  height: -moz-fit-content;
  height: fit-content;
  padding: 0 80px;
}

.c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  width: 22%;
  min-width: 250px;
  height: 197px;
  cursor: pointer;
  background: white;
  border: 1px solid;
  border-radius: 8px;
  margin: 1.5%;
  padding: 25px;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.c6 {
  font-weight: bold;
  font-size: 20px;
  line-height: 23px;
  text-align: start;
  margin: 0;
  color: black;
}

.c5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  width: 100%;
  height: -webkit-fit-content;
  height: -moz-fit-content;
  height: fit-content;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.c5 p {
  font-weight: bold;
  font-size: 16px;
  line-height: 19px;
  text-align: start;
  margin: 0;
  color: #898989;
}

.c5 img {
  cursor: pointer;
}

.c7 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-flex-wrap: nowrap;
  -ms-flex-wrap: nowrap;
  flex-wrap: nowrap;
  overflow: hidden;
  width: 100%;
  height: -webkit-fit-content;
  height: -moz-fit-content;
  height: fit-content;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
}

.c8 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: -webkit-fit-content;
  width: -moz-fit-content;
  width: fit-content;
  height: 28px;
  border-radius: 43px;
  margin-right: 10px;
  padding-left: 10px;
  padding-right: 10px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: rgba(255,255,255,0.53);
  white-space: nowrap;
  font-weight: bold;
  font-size: 12px;
  color: black;
}

<div
  className="c0"
>
  <div
    className="c1"
  >
    <div
      className="c2"
    >
      <h6>
        Sort By  
      </h6>
      <p>
        recommended
      </p>
      <img
        alt="sort"
        src="down-arrow.png"
      />
    </div>
  </div>
  <div
    className="c3"
  >
    <div
      className="c4"
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      style={
        Object {
          "borderColor": "#C9E1AA",
        }
      }
    >
      <div
        className="c5"
      >
        <p>
          TEST 1101
        </p>
        <img
          alt="save course"
          src="bookmark.svg"
        />
      </div>
      <div
        className="c6"
      >
        Intro to Design
      </div>
      <div
        className="c7"
      >
        <div
          className="c8"
          style={
            Object {
              "background": "#BAE6FF",
            }
          }
        >
          4 Credits
        </div>
        <div
          className="c8"
          style={
            Object {
              "background": "#FFE5A1",
            }
          }
        >
          Fall, Spring
        </div>
      </div>
    </div>
  </div>
</div>
`;
